src/golang/cmd/server/handler/connect_integration.go

```diff
@@ -209,5 +208,9 @@ func ValidateConfig(
 		return http.StatusInternalServerError, errors.Wrap(err, "Unable to connect integration.")
 	}
 
-	return http.StatusBadRequest, errors.Newf("Unable to authenticate credentials: %v", metadata.Error)
+	errMsg := ""
+	if metadata.Error != nil {
+		errMsg = metadata.Error.Context
```

Comment:
Might be better to throw a different internal server error here, since `execState.Error` being nil means something very wrong has happened right? Our assumption here is that the execution has definitely failed. An empty error string may be less helpful

So you can then directly return:
```
return http.StatusBadRequest, errors.Newf("Unable to auth...", execState.Error.Context)
```

Reply:
Good catch, we should return something more meaningful if the error is nil